File: core/src/main/java/com/toocol/ssh/core/shell/commands/processors/ShellLcCmdProcessor.java
package com.toocol.ssh.core.shell.commands.processors;

import com.toocol.ssh.core.shell.commands.ShellCommandProcessor;
import com.toocol.ssh.core.shell.core.Shell;
import com.toocol.ssh.utilities.utils.Tuple2;
import io.vertx.core.eventbus.EventBus;

import java.util.concurrent.atomic.AtomicBoolean;

/**
 * command: lc
 * desc:    list all the connection properties to quick switch
 *
 * @author ZhaoZhe (joezane.cn@gmail.com)
 * @date 2022/7/28 14:32
 */
public class ShellLcCmdProcessor extends ShellCommandProcessor {
    @Override
    public Tuple2<String, Long> process(EventBus eventBus, Shell shell, AtomicBoolean isBreak, String cmd) {
        return null;
    }
}

File: core/src/main/java/com/toocol/termio/core/shell/commands/processors/ShellLcCmdProcessor.kt
package com.toocol.termio.core.shell.commands.processors

import com.toocol.termio.core.cache.SWITCH_SESSION
import com.toocol.termio.core.shell.commands.ShellCommandProcessor
import com.toocol.termio.core.shell.core.Shell
import com.toocol.termio.utilities.utils.Tuple2
import io.vertx.core.eventbus.EventBus
import java.util.concurrent.atomic.AtomicBoolean

/**
 * command: lc
 * desc:    list all the connection properties to quick switch
 *
 * @author ZhaoZhe (joezane.cn@gmail.com)
 * @date 2022/7/28 14:32
 */
class ShellLcCmdProcessor : ShellCommandProcessor() {
    override fun process(
        eventBus: EventBus,
        shell: Shell,
        isBreak: AtomicBoolean,
        cmd: String
    ): Tuple2<String?, Long?> {
        val changeSession = shell.switchSession()
        isBreak.set(changeSession)
        SWITCH_SESSION = changeSession
        return Tuple2(null, shell.sessionId)
    }
}